test/CMakeLists.txt

```diff
@@ -67,6 +70,18 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE /bigobj)
     endif()
 
+    # Disable checked iterators on Windows for debug mode
+    # For details please see
+    #    https://learn.microsoft.com/en-us/cpp/standard-library/iterator-debug-level?view=msvc-170
+    #    https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
+    #    https://stackoverflow.com/questions/51494506/replacing-md-with-mt-in-cmake-is-not-possible-in-release-mode
+    if (WIN32 AND ${_build_type_in_lower} STREQUAL "debug")
+        if (${switch_off_checked_iterators})
+            target_compile_definitions(${_test_name} PRIVATE _ITERATOR_DEBUG_LEVEL=0)
+            target_compile_options(${_test_name} PRIVATE "/MD$<$<CONFIG:Debug>:>")
```

Comment:
I didn't find any other working ways to set this up for a specific application.

Reply:
Somewhat relevant to this discussion, but not a solution:
https://github.com/oneapi-src/oneDPL/pull/851#issuecomment-1536369021
in #851, we encountered a separate issue with icpx (GNU-style intel driver) using CMake on Windows which we cannot work around.  
If this is a blocker on icpx, this is more reason to push people to use icx[-cl] on Windows.

Reply:
I'm not saying that we shouldn't try to support GNU-style compilers (and it may be relevant for non-intel compilers also), but yes, this is perhaps less of an issue assuming that proposal goes through.

Reply:
I just tested this patch with icpx.exe, which is a GNU-style compiler and it works fine. It seems like it will work with and without ` -D_ITERATOR_DEBUG_LEVEL` defined and the compiler just ignores the `/MD` flag. So it's okay from that perspective.